src/span.ts

```diff
@@ -1,12 +1,14 @@
 import * as noop from './noop';
+import SpanContext from './span_context';
+import Tracer from './tracer';
 
 /**
  * Span represents a logical unit of work as part of a broader Trace. Examples
  * of span might include remote procedure calls or a in-process function calls
  * to sub-components. A Trace has a single, top-level "root" Span that in turn
  * may have zero or more child Spans, which in turn may have children.
  */
-export default class Span {
+export class Span {
```

Comment:
I don't really understand your concern. These classes _are_ already the noop implementation. What coupling?
For example, if you look at how I used them here: https://sourcegraph.com/github.com/sourcegraph/javascript-typescript-langserver/-/blob/src/project-manager.ts#L264:1-265:1
I declare each method to take a `Span` that defaults to a noop `Span` instance, and I can pass e.g. a LightStep `Span` just fine because it is compatible.

I totally agree that having all these underscore-prefixed methods is kind of useless, they serve no purpose besides little things like not having to convert `Span | SpanContext` to `SpanContext`. Implementations could just override the public methods. That would be a breaking change though.

What we could do is do a complete restructure, have all these now-classes be interfaces (those will not be translated to JS in any way), and then add classes like `NoopSpan`, `NoopTracer` etc. You would use them like this then:
```ts
import { Span, NoopSpan } from 'opentracing'
function foo(span: Span = new NoopSpan()) {

}